diff --git a/node_modules/karma-browserstack-launcher/index.js b/node_modules/karma-browserstack-launcher/index.js
index 4e61211..d874835 100644
--- a/node_modules/karma-browserstack-launcher/index.js
+++ b/node_modules/karma-browserstack-launcher/index.js
@@ -18,8 +18,10 @@ var createBrowserStackTunnel = function (logger, config, emitter) {
 
   log.debug('Starting BrowserStackLocal')
   bsLocal.start(bsLocalArgs, function () {
-    log.debug('Started BrowserStackLocal')
-    deferred.resolve()
+    setTimeout(function() {
+      log.debug('Started BrowserStackLocal')
+      deferred.resolve()
+    }, 2000);
   })
 
   emitter.on('exit', function (done) {
